src/components/pagination/get-pagination-classes.js

```diff
@@ -0,0 +1,73 @@
+const getPaginationClasses = (firstPage, currentPage, lastPage, pageElementClass) => {
+
+    let leftStepClass = pageElementClass;
+    let firstPageClass = pageElementClass;
+    let leftGapClass = pageElementClass;
+    let previousPageClass = pageElementClass;
+    let currentPageClass = pageElementClass + "active";
+    let nextPageClass = pageElementClass;
+    let rightGapClass = pageElementClass;
+    let lastPageClass = pageElementClass;
+    let rightStepClass = pageElementClass;
+
+
+    const gapLeap = Math.ceil(lastPage/10.0);
+    let rightGapLeap = gapLeap;
+    let leftGapLeap = gapLeap;
+    if(currentPage + gapLeap + 1 > lastPage){
+        rightGapLeap = lastPage - currentPage - 2;
+    }
+    if(currentPage - gapLeap - 1 < firstPage){
+        leftGapLeap = currentPage - firstPage - 2;
+    }
+    const noPages = typeof lastPage == "undefined" || lastPage === null || lastPage <= 0;
```

Comment:
if you use values returned by Unsplash server in its response (e.g. total_pages) , you will not need to do such a value check.